File: Python/BAEKJOON/BRONZE Ⅰ/14626 ISBN.py
# 손상된 ISBN 입력 (예: 9788968322*73)
original = input()

for i in range(10):
    # '*'를 0~9 숫자로 순서대로 대체
    ISBN = original.replace("*", str(i))

    # ISBN-13 가중합 계산
    # 짝수 인덱스(0,2,4...) × 1, 홀수 인덱스(1,3,5...) × 3
    total = sum(
        int(ISBN[j]) if j % 2 == 0 else int(ISBN[j]) * 3
        for j in range(13)
    )

    # 가중합이 10의 배수이면 유효한 ISBN → 해당 숫자 출력 후 종료
    if total % 10 == 0:
        print(i)
        break

File: javascript/BAEKJOON/BRONZE Ⅰ/14626 ISBN.js
var input = require('fs').readFileSync(__dirname+'/test.txt').toString().trim();
//var input = require('fs').readFileSync("/dev/stdin", "utf-8").trim();

const original = input;

for (let i = 0; i <= 9; i++) {
    const ISBN = original.replace("*", i);

    let sum = 0;
    for (let j = 0; j < 13; j++) {
        sum += j % 2 === 0? parseInt(ISBN.charAt(j)) : parseInt(ISBN.charAt(j)) * 3
    }

    if(sum % 10 === 0){
        console.log(i);
        break;
    }
}
